plat/common/arm/pauth.c

```diff
@@ -72,27 +28,40 @@ static inline bool pauth_supported(void)
 	apa = (reg >> ID_AA64ISAR1_EL1_APA_SHIFT) & ID_AA64ISAR1_EL1_APA_MASK;
 	api = (reg >> ID_AA64ISAR1_EL1_API_SHIFT) & ID_AA64ISAR1_EL1_API_MASK;
 	if (apa || api)
-		return true;
+		return __true;
 
 	reg = SYSREG_READ(ID_AA64ISAR2_EL1);
 	apa3 = (reg >> ID_AA64ISAR2_EL1_APA3_SHIFT) & ID_AA64ISAR2_EL1_APA3_MASK;
 	if (apa3)
-		return true;
+		return __true;
 
-	return false;
+	return __false;
 }
 
 int __no_pauth ukplat_pauth_init(void)
 {
 	__u64 key_hi, key_lo;
 	__u64 reg;
+	int rc;
 
-	if (unlikely(!pauth_supported()))
+	if (unlikely(!pauth_supported())) {
+		uk_pr_err("The CPU does not implement FEAT_PAUTH\n");
```

Comment:
I have a similar comment here regarding the message as I mentioned to the x86 `RDRAND` feature regarding "implement". Is it possible that features can be masked out on Arm for virtualized guests?

Reply:
Yes, on `arm64` the hypervisor can trap the `mrs` to the feature identification registers and return a custom value to the guest. For the rest let's continue the discussion in https://github.com/unikraft/unikraft/pull/1496#discussion_r1858301885.

Reply:
Okay, then the same answer in that comment applies here :)